### PR TITLE
Add tab width modes: equal and titleLength

### LINKED ISSUE #597: Tab sizing
Initially brought up in #444, the ability to enable/disable a configurable min-width for tabs when you have many of them, similar to:

If disabled, it would be chrome-like with tabs getting ever smaller, and if enabled, it would be firefox-like with a user-defined min-width before it starts either scrolling or being paginated.

### PR DESCRIPTION
**Pull request by cinnamon-msft**
## Summary of the Pull Request
Created new global setting "tabWidthMode"
Will accept "equal" or "titleLength"
Equal: Similar to browser experience where all tabs are equal width
TitleLength: Original implementation where width will fit whole tab title

Changed default width behavior to Equal

Refs #597 

terminal-equal-tabs-pr

### CODE REVIEW

File: doc/cascadia/SettingsSchema.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ Properties listed below affect the entire window, regardless of the profile sett
 | `requestedTheme` | _Required_ | String | `system` | Sets the theme of the application. Possible values: `"light"`, `"dark"`, `"system"` |
 | `showTerminalTitleInTitlebar` | _Required_ | Boolean | `true` | When set to `true`, titlebar displays the title of the selected tab. When set to `false`, titlebar displays "Windows Terminal". |
 | `showTabsInTitlebar` | Optional | Boolean | `true` | When set to `true`, the tabs are moved into the titlebar and the titlebar disappears. When set to `false`, the titlebar sits above the tabs. |
+| `tabWidth` | Optional | String | `equal` | Sets the width of the tabs. Possible values: `"equal"`, `"sizetocontent"` |
```
Comment (carlos-zamora) [inline doc/cascadia/SettingsSchema.md]:
```suggestion
| `tabWidth` | Optional | String | `equal` | Sets the width of the tabs. Possible values: `"equal"`, `"sizeToContent"` |
```
I think everything else is camel case, so we should probably match that.

**Comment by DHowett-MSFT** [comment]
* should it be profile-specific?

**Comment by DHowett-MSFT** [comment]
* should it live at the root object? what if we _do_ want to make mode and min/max separate things?

**Comment by rkeithhill** [comment]
RE `equal` vs `size-to-content`, what I'd like to see isn't necessarily either.  Ideally, what I want is a tab system that obeys this one simple precept: all tabs are always visible with no need to scroll to see tabs.  Well, except when perhaps the tabs get ridiculously narrow e.g.:

image

But ^ is fairly corner case.  

Given this particularly precept, when there are a small number of tabs, you could size to content but when those combined tab widths exceed the width of the tab area, then you switch to fixed-width such that all tabs are guaranteed to be visible.

File: doc/cascadia/SettingsSchema.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ Properties listed below affect the entire window, regardless of the profile sett
 | `requestedTheme` | _Required_ | String | `system` | Sets the theme of the application. Possible values: `"light"`, `"dark"`, `"system"` |
 | `showTerminalTitleInTitlebar` | _Required_ | Boolean | `true` | When set to `true`, titlebar displays the title of the selected tab. When set to `false`, titlebar displays "Windows Terminal". |
 | `showTabsInTitlebar` | Optional | Boolean | `true` | When set to `true`, the tabs are moved into the titlebar and the titlebar disappears. When set to `false`, the titlebar sits above the tabs. |
+| `tabWidth` | Optional | String | `equal` | Sets the width of the tabs. Possible values: `"equal"`, `"sizeToContent"` |
```
Comment (zadjii-msft) [inline doc/cascadia/SettingsSchema.md]: This doc suggests `sizeToContent`, though the rest of the pr says `sizetocontent`. I'm pretty sure we should go with the camelCase one

**Comment by zadjii-msft** [comment]
> 
>     * should it live at the root object? what if we _do_ want to make mode and min/max separate things?

I think I'm okay with this being in the root (globals) for now. This is an "application" settings, so it should live with the other application settings. In the future, we might want to consider these properties as being a part of the 
[_theme_](https://github.com/microsoft/terminal/issues/3327), but I don't think we should hold up putting them in the root for now over that. We can always do a similar colorscheme/foreground (where the property from the profile overrides that from the colorscheme) thing with this once themes exist.

**Comment by vadimkantorov** [comment]
So many people raised issues, I hope equal mode is chosen as the default one

**Comment by zadjii-msft** [comment]
_shh msftbot, it's the holidays. Take a break._

**Comment by rkeithhill** [comment]
msftbot needs to be made aware of extended holiday breaks.  :-)

**Comment by miniksa** [comment]
This comment represents me seeing this PR but choosing to not review it because there's already 3 cooks in this kitchen (despite the holiday). It serves to remind me of this decision in the future.

**Comment by cinnamon-msft** [comment]
Spec can be found here: #4104

**Pull request merged**